kubeflow/hub/api/model_registry_client.py

```diff
@@ -0,0 +1,285 @@
+        metadata: Mapping[str, SupportedTypes] | None = None,
+    ) -> RegisteredModel:
+        """Register a model.
+
+        This registers a model in the model registry. The model is not downloaded,
+        and has to be stored prior to registration.
+
+        Most models can be registered using their URI, along with optional
+        connection-specific parameters, `storage_key` and `storage_path` or,
+        simply a `service_account_name`. URI builder utilities are recommended
+        when referring to specialized storage; for example `utils.s3_uri_from`
+        helper when using S3 object storage data connections.
+
+        Args:
+            name: Name of the model.
+            uri: URI of the model.
+
+        Keyword Args:
+            version: Version of the model. Has to be unique.
+            model_format_name: Name of the model format.
+            model_format_version: Version of the model format.
+            description: Description of the model.
+            author: Author of the model. Defaults to the client author.
+            owner: Owner of the model. Defaults to the client author.
+            storage_key: Storage key.
+            storage_path: Storage path.
+            service_account_name: Service account name.
+            model_source_kind: A string identifier describing the source kind.
+            model_source_class: A subgroup within the source kind.
+            model_source_group: This identifies a source group for models from source class.
+            model_source_id: A unique identifier for a source model within kind, class, and group.
+            model_source_name: A human-readable name for the source model.
+            metadata: Additional version metadata.
+
+        Returns:
+            Registered model.
+        """
+        return self._registry.register_model(
+            name=name,
+            uri=uri,
+            model_format_name=model_format_name,
+            model_format_version=model_format_version,
+            version=version,
+            storage_key=storage_key,
+            storage_path=storage_path,
+            service_account_name=service_account_name,
+            model_source_kind=model_source_kind,
+            model_source_class=model_source_class,
+            model_source_group=model_source_group,
+            model_source_id=model_source_id,
+            model_source_name=model_source_name,
+            author=author,
+            owner=owner,
+            description=description,
+            metadata=metadata,
+        )
+
+    def update_model(self, model: RegisteredModel) -> RegisteredModel:
+        """Update a registered model.
+
+        Args:
+            model: The registered model to update. Must have an ID.
+
+        Returns:
+            Updated registered model.
+
+        Raises:
+            model_registry.exceptions.StoreError: If model does not have an ID.
+        """
+        return self._registry.update(model)
```

Comment:
This `update_*` methods are effectively all aliases (aside from type-annotations) - maybe worth actually erroring if a user calls the wrong one? For example, we could throw an exception if a user passes a `ModelVersion` to `update_model` instead of to `update_model_version`

Reply:
+1 thanks for following up on the original discussion in https://github.com/kubeflow/sdk/issues/59#issuecomment-3564188759